doc/en/user/source/community/cog/index.rst

```diff
@@ -0,0 +1,108 @@
+.. _community_cog:
+
+COG (Cloud Optimized GeoTIFF) Support
+=====================================
+
+`COG <https://github.com/cogeotiff/cog-spec/blob/master/spec.md>`_ (Cloud Optimized GeoTIFF) is a regular GeoTIFF file, aimed at being hosted on a HTTP file server, whose internal organization is friendly for consumption by clients issuing HTTP GET range requests. 
+The COG module allows to set configuration params to connect to a Cloud GeoTIFF as well as adding JARs to the classpath needed to support that.
+
+Installation
+------------
+
+As a community module, the package needs to be downloaded from the `nightly builds <https://build.geoserver.org/geoserver/>`_,
+picking the community folder of the corresponding GeoServer series (e.g. if working on the GeoServer main development branch nightly
+builds, pick the zip file form ``master/community-latest``).
+
+To install the module, unpack the zip file contents into GeoServer own ``WEB-INF/lib`` directory and restart GeoServer.
+
+COG GeoTIFF Configuration Panel
+-------------------------------
+
+When configuring a GeoTIFF store, a new checkbox is available: ``Cloud Optimized GeoTIFF (COG)``. Setting that will open a new section presenting the COG configuration parameters for this COG Store.
+
+.. figure:: images/cogparams.png
+
+   COG Connection params
+
+Checking the ``Cloud Optimized GeoTIFF (COG)`` checkbox will open 
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Option
+     - Description
+   * - :guilabel:`URL`
+     - (prefixed by ``cog://``) representing the connection URL to the COG Dataset.
+   * - :guilabel:`Range Reader Settings`
+     - Which type of Range Reader implementation. Values Currently supported are HTTP and S3, the latter using an S3 Client
```

Comment:
Looking at this made me think about the GWC S3 support, and putting both toghether in the same GeoServer.
They are not using the same version of the libraries:
* https://github.com/geosolutions-it/imageio-ext/blob/master/plugin/cog/pom.xml#L18
* https://github.com/GeoWebCache/geowebcache/blob/master/geowebcache/s3storage/pom.xml#L22

I know someone has looked into upgrading the AWS SDK in GWC, but v2 of the library has a signficantly different API, requiring deep changes in the module. 
I don't know if the two can stay in the same classpath (may be they are using different packages), but if they don't, there should be at least a warning here.

Reply:
Thanks for spotting that.
As soon as I get the final zip with all the needed dependencies I'll try a local deploy of both pieces and check how it works.
As far as I can see they use different packages 

      <groupId>com.amazonaws</groupId>
      <artifactId>aws-java-sdk-s3</artifactId>
vs 
       <groupId>software.amazon.awssdk</groupId>
       <artifactId>s3</artifactId>

It "seems" they can live together:
https://docs.aws.amazon.com/sdk-for-java/v2/migration-guide/side-by-side.html

Anyway, an actual-practical check will let us know :)  
I'll keep you updated on the findings.

Reply:
@aaime update:
I managed to finalize the ext-cog.xml containing aaaaall the needed jars to build the zip file.
I have also deployed both cog and gwc-s3 module and they works together. 
I have tested a new s3 blobstore and played a bit with tiled layer preview on a cog layer: the cog layer was rendering fine and the S3 bucket get populated with new JPEG tile files so I think we are good with this double check.